src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    if (this.length === 0) {
+      return strJoin;
+    }
+
+    for (let value of this) {
+      if (this.length === 1) {
+        strJoin += value;
+
+        return strJoin;
+      }
```

Comment:
```suggestion
```

No need for this check. In your code below you always add value to your string (if the length of the array is 1 or more)